src/main/java/service/TransactionParser.java

```diff
@@ -0,0 +1,41 @@
+package service;
+
+import dto.Transaction;
+import lombok.val;
+import model.OperationEnum;
+
+import java.util.List;
+
+public class TransactionParser {
+
+    private static final String DATA_SEPARATOR = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int PRODUCT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final int NORMAL_ACTIVITY_LENGTH = 3;
+
+
+    public static List<Transaction> parseToTransaction(List<String> storeActivities) {
+
+        return storeActivities.stream()
+                .map(TransactionParser::splitActivity)
+                .map(TransactionParser::createTransaction)
+                .toList();
+    }
+
+    private static String[] splitActivity(String activity){
+
+        val splitActivityArray = activity.strip().split(DATA_SEPARATOR);
+        if (splitActivityArray.length != NORMAL_ACTIVITY_LENGTH) {
+            throw new IllegalArgumentException();
```

Comment:
It's always better to use custom message when you throw an exception, so the user will understand what actually gone wrong

Reply:
Thanks for your wonderful advice, done!